c7n/actions/autoscaling.py

```diff
@@ -0,0 +1,254 @@
+    """Action to resize the min/max/desired count in an application autoscaling target
+
+    There are several ways to use this action:
+
+    1. apply a fixed resize of min, max or desired, optionally saving the
+       previous values to a named tag (for restoring later):
+
+    .. code-block:: yaml
+
+            policies:
+              - name: offhours-ecs-off
+                resource: ecs-service
+                filters:
+                  - type: offhour
+                    offhour: 19
+                    default_tz: bst
+                actions:
+                  - type: resize
+                    min-capacity: 0
+                    desired: 0
+                    save-options-tag: OffHoursPrevious
+                    suspend-scaling: true
+
+    2. restore previous values for min/max/desired from a tag:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: offhours-ecs-on
+                resource: ecs-service
+                filters:
+                  - type: onhour
+                    onhour: 8
+                    default_tz: bst
+                actions:
+                  - type: resize
+                    restore-options-tag: OffHoursPrevious
+                    restore-scaling: true
+
+    """
+
+    schema = type_schema(
+        'resize',
+        **{
+            'min-capacity': {'type': 'integer', 'minimum': 0},
+            'max-capacity': {'type': 'integer', 'minimum': 0},
+            'desired': {
+                "anyOf": [
+                    {'enum': ["current"]},
+                    {'type': 'integer', 'minimum': 0}
+                ]
+            },
+            'save-options-tag': {'type': 'string'},
+            'restore-options-tag': {'type': 'string'},
+            'suspend-scaling': {'type': 'boolean'},
+            'restore-scaling': {'type': 'boolean'},
+        }
+    )
+    autoscaling_permissions = (
+        'application-autoscaling:DescribeScalableTargets',
+        'application-autoscaling:RegisterScalableTarget',
+    )
+
+    def get_permissions(self):
+        return self.autoscaling_permissions + self.permissions
+
+    @property
+    def scalable_dimension(self):
+        """ the scalable dimension for the Application Autoscaling target """
+
+    @property
+    def service_namespace(self):
+        """ the service namespace for interacting with Application Autoscaling """
+
+    def get_resource_id(self, resource):
+        """ return the id for the provided resource """
+        raise NotImplementedError
+
+    def get_resource_tag(self, resource, key):
+        """ return the tag for the provided resource """
+        raise NotImplementedError
+
+    def get_resource_desired(self, resource):
+        """ return the current desired value for the provided resource """
+        raise NotImplementedError
+
+    def set_resource_tag(self, resource, key, value):
+        """ set the tag for the provided resource """
+        raise NotImplementedError
```

Comment:
we can process this capability generic for any resource that supports tagging, ie thats basically how auto-tagging user works. that then makes get_resource_id extraneous.

Reply:
I believe I'll still need the `get_resource_id` for other reasons, but i like the idea of getting rid of `set_resource_tag` and replacing with `UniversalTag`. I've pushed a commit with that implementation.